api/src/utils/permissions-cachable.ts

```diff
@@ -0,0 +1,44 @@
+import type { Accountability, Filter } from '@directus/types';
+import { fetchPermissions } from '../permissions/lib/fetch-permissions.js';
+import { fetchPolicies } from '../permissions/lib/fetch-policies.js';
+import type { Context } from '../permissions/types.js';
+import { createDefaultAccountability } from '../permissions/utils/create-default-accountability.js';
+
+/**
+ * Check if the read permissions for a collection contain the dynamic variable $NOW.
+ * If they do, the permissions are not cachable.
+ */
+export async function permissionsCachable(collection: string, context: Context, accountability?: Accountability) {
+	if (!accountability) {
```

Comment:
Due to `req.collection` not being defined for all endpoints should we early return for those case (e.g. `POST /auth/login`)?

Reply:
At least according to our typescript definitions, req.collection should always be defined, that's why I ignored this case.

Reply:
The example given with `POST /auth/login` has it as undefined so the typing is incorrect

Reply:
Resolved via https://github.com/directus/directus/pull/24543/commits/f254de6b3a3cefe2d8aa6239fb4a24c0d676abd0